account-compression/programs/spl-compression/src/data_wrapper/mod.rs

```diff
@@ -0,0 +1,30 @@
+//! # Data Wrapper
+//! We use CPI calls to circumvent the 10kb log limit on Solana transactions.
+//! Instead of logging events to the runtime, we execute a CPI to the `wrapper` program
+//! where the log data is serialized into the instruction data.
+//!
+//! This works because CPI instruction data is never truncated. Logging information is
+//! vital to the functioning of compression. When compression logs are truncated, indexers can fallback to 
+//! deserializing the CPI instruction data.
```

Comment:
Maximum number of bytes needed is `32 * (MAX_DEPTH + 1) == 32 * 32 == 1024` per operation.

It might be possible to slim this down (cc @jarry-xiao )

Reply:
Ah ok, how about this: transaction return data is set to be exactly 1024 bytes, and should be logged by validators.  The only issue is that you can run into an issue if another program overwrites it.  Would you be able to use that?

Reply:
Practically inconvenient to wait for the runtime to implement this new solution. I agree that this is a hack, but logs getting dropped silently should be 100% unacceptable IMO. I would be more okay with building a separate undroppable log buffer or somehow injecting a Noop program in every VM so the CPI overhead is as low as possible. I’m pretty sure Ethereum does not drop events

Reply:
Also I think we need a little more data that that @ngundotra (Node index, tree_id). It’s always less than the 1280 byte CPI instruction size limit so there’s no reason to worry unless that changes.

Reply:
Return data is currently available in the runtime on all networks if you just use `program::set_return_data` https://github.com/solana-labs/solana/blob/9703855a80d6f5991a093c98c626cbbdb7e628e1/sdk/program/src/program.rs#L322 and validators can log if they choose with the `--enable-extended-tx-metadata-storage` flag.

Either way, the more I think about it, that would just be another hack.  How many bytes can you log currently with normal `msg`?

Reply:
Yeah I've known about the return data feature for a while, but ultimately we went with the option that seemed to show up most consistently in RPC calls. The problem with msg is that I can prepend a transaction to maliciously overflow the log buffer and mess up the indexer, so even logging a single byte would be unsafe